src/networking.rs

```diff
@@ -23,6 +24,12 @@ const MAX_PEER_ID_BYTE_COUNT: usize = 512;
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct PeerId(#[serde(with = "crate::serde::as_b58")] ByteList<MAX_PEER_ID_BYTE_COUNT>);
 
+impl fmt::Display for PeerId {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.0)
```

Comment:
as written, this will just call `Display::fmt` on the inner `ByteList` which I think will just dump the hex-encoded byte list

but instead we want the "canonical" base58 encoding so will want to do something like this

```suggestion
#[cfg(feature = "serde")]
impl fmt::Display for PeerId {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "{}", bs58::encode(self.0))
```

Reply:
hmm actually this also is wrapped in a "multi hash"

you can see the implementation here:

https://github.com/libp2p/go-libp2p/blob/v0.23.2/core/peer/peer.go#L179

I'd suggest looking for a `multihash` impl or referring to a rust libp2p impl to see how they implement this and just copy that code